com.unity.multiplayer.mlapi/Prototyping/NetworkTransform.cs

```diff
@@ -77,70 +184,154 @@ public void NetworkSerialize(NetworkSerializer serializer)
+
+            return
+                // InLocalSpace Check
+                (InLocalSpace != networkState.InLocalSpace) ||
+                // Position Check
+                (SyncPositionX && !Mathf.Approximately(position.x, networkState.PositionX)) ||
+                (SyncPositionY && !Mathf.Approximately(position.y, networkState.PositionY)) ||
+                (SyncPositionZ && !Mathf.Approximately(position.z, networkState.PositionZ)) ||
+                // RotAngles Check
+                (SyncRotAngleX && !Mathf.Approximately(rotAngles.x, networkState.RotAngleX)) ||
+                (SyncRotAngleY && !Mathf.Approximately(rotAngles.y, networkState.RotAngleY)) ||
+                (SyncRotAngleZ && !Mathf.Approximately(rotAngles.z, networkState.RotAngleZ)) ||
+                // Scale Check
+                (SyncScaleX && !Mathf.Approximately(scale.x, networkState.ScaleX)) ||
+                (SyncScaleY && !Mathf.Approximately(scale.y, networkState.ScaleY)) ||
+                (SyncScaleZ && !Mathf.Approximately(scale.z, networkState.ScaleZ));
+        }
+
+        internal void UpdateNetworkState()
+        {
+            var position = InLocalSpace ? m_Transform.localPosition : m_Transform.position;
+            var rotAngles = InLocalSpace ? m_Transform.localEulerAngles : m_Transform.eulerAngles;
+            var scale = InLocalSpace ? m_Transform.localScale : m_Transform.lossyScale;
+
+            // InLocalSpace Bit
+            ReplNetworkState.Value.InLocalSpace = InLocalSpace;
+            // Position Bits
+            (ReplNetworkState.Value.HasPositionX, ReplNetworkState.Value.HasPositionY, ReplNetworkState.Value.HasPositionZ) =
+                (SyncPositionX, SyncPositionY, SyncPositionZ);
+            // RotAngle Bits
+            (ReplNetworkState.Value.HasRotAngleX, ReplNetworkState.Value.HasRotAngleY, ReplNetworkState.Value.HasRotAngleZ) =
+                (SyncRotAngleX, SyncRotAngleY, SyncRotAngleZ);
+            // Scale Bits
+            (ReplNetworkState.Value.HasScaleX, ReplNetworkState.Value.HasScaleY, ReplNetworkState.Value.HasScaleZ) =
+                (SyncScaleX, SyncScaleY, SyncScaleZ);
+
+            // Position Values
+            (ReplNetworkState.Value.PositionX, ReplNetworkState.Value.PositionY, ReplNetworkState.Value.PositionZ) =
+                (position.x, position.y, position.z);
+            // RotAngle Values
+            (ReplNetworkState.Value.RotAngleX, ReplNetworkState.Value.RotAngleY, ReplNetworkState.Value.RotAngleZ) =
+                (rotAngles.x, rotAngles.y, rotAngles.z);
+            // Scale Values
+            (ReplNetworkState.Value.ScaleX, ReplNetworkState.Value.ScaleY, ReplNetworkState.Value.ScaleZ) =
+                (scale.x, scale.y, scale.z);
+
+            ReplNetworkState.SetDirty(true);
+        }
+
+        // TODO: temporary! the function body below probably needs to be rewritten later with interpolation in mind
```

Comment:
I'd put this onto your plate @SamuelBellomo when you start working on interpolator integration & polish.